internal/discover/graphics.go

```diff
@@ -74,17 +77,75 @@ func NewGraphicsMountsDiscoverer(logger logger.Interface, driver *root.Driver, n
 		},
 	)
 
+	symlinks := newGraphicsDriverSymlinks(logger, libraries, nvidiaCDIHookPath)
 	xorg := optionalXorgDiscoverer(logger, driver, nvidiaCDIHookPath)
 
 	discover := Merge(
 		libraries,
 		jsonMounts,
+		symlinks,
 		xorg,
 	)
 
 	return discover, nil
 }
 
+type graphicsDriverSymlinks struct {
+	None
+	logger            logger.Interface
+	libraries         Discover
+	nvidiaCDIHookPath string
+}
+
+var _ Discover = (*graphicsDriverSymlinks)(nil)
+
+func newGraphicsDriverSymlinks(logger logger.Interface, libraries Discover, nvidiaCDIHookPath string) Discover {
+	return &graphicsDriverSymlinks{
+		logger:            logger,
+		libraries:         libraries,
+		nvidiaCDIHookPath: nvidiaCDIHookPath,
+	}
+}
+
+// Create necessary library symlinks for graphics drivers
+func (d graphicsDriverSymlinks) Hooks() ([]Hook, error) {
+	mounts, err := d.libraries.Mounts()
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get library mounts: %v", err)
+	}
+
+	links := []string{}
+
+	for _, mount := range mounts {
+		filename := filepath.Base(mount.HostPath)
+
+		// nvidia-drm_gbm.so is a symlink to libnvidia-allocator.so
+		// Make sure this is actually available
+		if strings.HasPrefix(filename, "libnvidia-allocator.so.") {
+			linkDir := filepath.Dir(mount.Path)
+			linkPath := filepath.Join(linkDir, "gbm", "nvidia-drm_gbm.so")
+			target := filepath.Join("..", filename)
+			links = append(links, fmt.Sprintf("%s::%s", target, linkPath))
```

Comment:
@elezar I changed this to a symlink hook, but I'm not sure if this should be done this way.

When doing it via `NewMounts` as before, the locator will attempt to resolve the symlink on the host side. That means the path gets only injected when the library actually exists on the host and future changes to this path (link target changes, becoming a regular file, etc...) will also be covered.

So I'd advocate to put this back to `NewMounts` as originally suggested by @ehfd 

Your Thoughts?

Reply:
This depends on @elezar 's policies on this matter.

Reply:
I don't think that adding `gbm/nvidia-drm_gbm.so` to the `NewMounts` will have the desired effect. Since this resolves to `libnvidia-allocator.so` we will only include `libnvidia-allocator.so` in the list of mounts and NOT create the symlink.

Note that you comment on handling changes in this path is not 100% accurate. While this is true for `gbm/nvidia-drm_gbm.so`, changing this to a regular file would have the downside that we may lose the resolution of the target library in this case.

Your comment does raise an interesting question though. If `gbm/nvidia-drm_gbm.so` exists on the host, then we could follow a similar strategy to what we do for Tegra-based systems [here](https://github.com/NVIDIA/nvidia-container-toolkit/blob/dfc8e22e12fccde4afcde6ce21f2fc84a64d98ec/internal/platform-support/tegra/csv.go#L62-L68). 

Where we distinguish between regular libraries and symlinks. In both cases any symlinks are followed to ensure that we only mount regular files into a container, but in the case of symlinks we recreate the entire symlink chain in the container. If we were to factor out this logic to be reused here, we could treat the symlinks here explicitly as separate entities to discover.

I do think that thsi can be done as a follow-up to this PR though.

Reply:
I agree that merging first is better since this is taking such a long time.

@elezar So are you satisfied with his changes?

Reply:
@ehfd I have just tried to push to your branch, but I'm getting permission errors.

I have rebased your branch off `main` and created https://github.com/ehfd/nvidia-container-toolkit/pull/1/files with my proposed changes.

Reply:
@elezar I added you to the repository for write access. Merge when you're ready.